### ISSUE TITLE
An error occured: Runner version v2.319.0 is deprecated and cannot receive messages.

### ISSUE THREAD

**Issue by quarckster** [actions/runner#3471]
**Describe the bug**
Runner v2.319.0 cannot get jobs from github.com on Ubuntu 22.04 arm64. 

## Runner Version and Platform
Version of your runner?
2.319.0

OS of the machine running the runner?
Ubuntu 22.04 arm64

## Job Log Output
```
--------------------------------------------------------------------------------
|        ____ _ _   _   _       _          _        _   _                      |
|       / ___(_) |_| | | |_   _| |__      / \   ___| |_(_) ___  _ __  ___      |
|      | |  _| | __| |_| | | | | '_ \    / _ \ / __| __| |/ _ \| '_ \/ __|     |
|      | |_| | | |_|  _  | |_| | |_) |  / ___ \ (__| |_| | (_) | | | \__ \     |
|       \____|_|\__|_| |_|\__,_|_.__/  /_/   \_\___|\__|_|\___/|_| |_|___/     |
|                                                                              |
|                       Self-hosted runner registration                        |
|                                                                              |
--------------------------------------------------------------------------------
# Authentication
√ Connected to GitHub
# Runner Registration
√ Runner successfully added
√ Runner connection is good
# Runner settings
√ Settings Saved.
√ Connected to GitHub
Current runner version: '2.319.0'
2024-09-20 14:09:09Z: Listening for Jobs
An error occured: Runner version v2.319.0 is deprecated and cannot receive messages.
Runner listener exit with terminated error, stop the service, no retry needed.
Exiting runner...
```

**Comment by den-is** [comment]
I've got the same issue and it caused significant downtime for my company with dozens of jobs.
Error message about `2.319.0 deprecated` is ridiculous!
It worked for months since release and why this fatal even happened today during the Friday (**night** for me)? (ofc all these questions are addressed to github runners/actions maintainers)

`2.319.1` was released just a few days after 2.319.0 with some minor changelog.
From my experience, I remember runners who were months and years old still running without any issues.

Also, I'm running actions-scaleset-controller - which is not updating runners-scalesets completely smoothly.

I'm curious why did you close this issue @quarckster?

**Comment by den-is** [comment]
Yea I upgraded too.
But the case is ridiculous, and this is not how things should be usually deprecated/handled/cut etc...
Nobody would say that 2.319.0 is too old and I can't see any message/warning about immediate upgrade.
I would like to hear an official explanation from maintainers.
